app/src/ts/video-model.ts

```diff
@@ -1,26 +1,228 @@
+       } from "safe-launcher-client";
+
+import { safeClient } from "./util";
+import * as fs from "fs";
+import * as fileType from "file-type";
+import * as readChunk from "read-chunk";
+import * as stream from "stream";
+
+import Config from "./global-config";
+const CONFIG: Config = Config.getInstance();
+
+class UnsupportedVideoFormatError extends Error {
+    constructor(mimeType: string) {
+        super(`Unsupported Video Format: ${mimeType}`);
+    }
+}
+
+export class Video implements Drop {
+
+    public readonly title: string;
+    public readonly description: string;
+    public readonly file: Promise<string>; // TODO: verify that there isn't a filepath type
+    public readonly owner: string;
+
+    public readonly commentReplies: AppendableDataHandle;
+    public readonly videoReplies: AppendableDataHandle;
+
+    private readonly commentMetadata: Promise<AppedableDataMetadata>;
+    private readonly videoRepliesMetadata: Promise<AppedableDataMetadata>;
+
+    constructor(title: string, description: string, owner: string, file: Promise<string>,
+                commentReplies: AppendableDataHandle, videoReplies: AppendableDataHandle) {
+        this.title = title;
+        this.description = description;
+        this.owner = owner;
+        this.file = file;
+
+        this.commentReplies = commentReplies;
+        this.videoReplies = videoReplies;
+
+        this.commentMetadata = commentReplies.getMetadata();
+        this.videoRepliesMetadata = videoReplies.getMetadata();
+    }
+    public async drop(): Promise<void> {
+        await this.commentReplies.drop();
+        await this.videoReplies.drop();
+    }
+
+    /**
+     *  Construct a new video from raw parts.
+     *
+     *  TODO: Once we have the DNS api working, this method needs to do the
+     *  right thing when it comes to setting the owner field with a long name or something.
+     *
+     * @returns a new Video. The video has not been persisted to the SAFEnet until `write` is called
+     */
+    public static async new(title: string, description: string, localVideoFile: string): Promise<Video> {
+        // Ownership of these guys is going to be passed to the created video
+        const commentReplies: AppendableDataHandle = await safeClient.ad.create(title + " commentReplies");
+        const videoReplies: AppendableDataHandle = await safeClient.ad.create(title + " videoReplies");
+
+        return new Video(title, description, "TODO OWNER",
+                         Promise.resolve(localVideoFile),
+                         commentReplies, videoReplies);
+    }
+
+    // @returns a promise for a dataID pointing to the written video meta-node
+    public async write(): Promise<DataIDHandle> {
+        await this.commentReplies.save().catch(err => {
+            // if the appendable data already exists, ignore the error. We don't need
+            // to update it.
+            if ((typeof err.res.statusCode !== "undefined" && err.res.statusCode === 400)
```

Comment:
These comparisons will work fine if you just do `err.res.statusCode === 400`.  If we did still want to check for existence too, I think that `err.res.statusCode == null` is the idiomatic way to do that.